spec/spec_helper.rb

```diff
@@ -14,24 +14,27 @@
   end
 end
 
+Object.include Spec::Shared::CassetteSecretsHelper
 VCR.configure do |config|
+  # config.default_cassette_options = { :record => :all }
+
   config.ignore_hosts 'codeclimate.com' if ENV['CI']
   config.cassette_library_dir = File.join(ManageIQ::Providers::Vmware::Engine.root, 'spec/vcr_cassettes')
 
-  secrets = Rails.application.secrets
-  config.define_cassette_placeholder(Rails.application.secrets.vmware_infra_defaults[:hostname]) do
-    Rails.application.secrets.vmware_infra[:hostname]
+  config.define_cassette_placeholder(default_vcr_secret_by_key_path(:vmware_infra, :hostname)) do
+    vcr_secret_by_key_path(:vmware_infra, :hostname)
```

Comment:
and this becomes

```suggestion
  config.define_cassette_placeholder(default_vcr_secret(:vmware_infra, :hostname)) do
    vcr_secret(:vmware_infra, :hostname)
```

which could possibly be inlined now that it's less wordy.